modules/contact/src/MechanicalContactConstraint.C

```diff
@@ -126,8 +129,19 @@ MechanicalContactConstraint::MechanicalContactConstraint(const InputParameters &
     _mesh_dimension(_mesh.dimension()),
     _vars(3, libMesh::invalid_uint),
     _var_objects(3, nullptr),
+    _has_slave_gap_offset(isParamValid("slave_gap_offset")),
+    _slave_gap_offset_var(
+        _has_slave_gap_offset
+            ? &_fe_problem.getStandardVariable(_tid, getParam<VariableName>("slave_gap_offset"))
+            : NULL),
+    _has_mapped_master_gap_offset(isParamValid("mapped_master_gap_offset")),
+    _mapped_master_gap_offset_var(
+        _has_mapped_master_gap_offset
+            ? &_fe_problem.getStandardVariable(_tid,
+                                               getParam<VariableName>("mapped_master_gap_offset"))
+            : NULL),
     _nodal_area_var(getVar("nodal_area", 0)),
-    _aux_system(_nodal_area_var->sys()),
+    _aux_system(_fe_problem.getAuxiliarySystem()),
```

Comment:
This is actually going to change the `_aux_system` from being the aux system corresponding to whether this object is being computed on the undisplaced or displaced mesh to ***always*** being the aux system of the undisplaced mesh. Is that what you want?

Reply:
I think for the gap offset variables, their values are preset and don't depend on the mesh being undisplaced or displaced. But for the _nodal_area_var, I think it use the displaced mesh for accurate nodal area calculation. I had segmentation fault using aux system got from _nodal_area_var for gap offset variables.